Fatras/include/ActsFatras/Digitization/ChannelMerger.hpp

```diff
@@ -0,0 +1,65 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Surfaces/SurfaceError.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+#include "ActsFatras/Digitization/DigitizationData.hpp"
+
+#include <array>
+#include <utility>
+
+namespace ActsFatras {
+
+/// Generic implementation of a channel merger, currently only additive
+/// channel merging
+///
+/// @tparam signal_t The type of signal, needs operator+= to be defined
+/// @tparam kSize the dimensonality of the object (cluster)
+///
+/// @param channels The channels from one cluster
+///
+/// @return A cluster containing the parameter set and cluster size
+template <typename signal_t, size_t kSize>
+const std::vector<Channel<signal_t, kSize>> mergeChannels(
+    const std::vector<Channel<signal_t, kSize>>& channels) {
```

Comment:
Could the channels be merged in-place? I do not know how this is going to be called in the end, but I suspect that it will be more performant to not create a copy be default, i.e. have the following interface instead:
```cpp
void mergeChannels(std::vector<Channel<signal_t, kSize>>&)
```
if you still want a copy, the user call site can create if manually, i.e.
```cpp
auto copy = input;
mergeChannels(copy);
```

To avoid having to rewrite the implementation, a temporary copy can be made within the function. Then we can replace the implementation later while keeping the interface.

Reply:
Good question, I haven't played that through, but at the end this is a rather rare case, so I think a copy will be fine in all cases.

Reply:
I would suggest we leave it for the moment, as it gives a more clearer input/output structure.

Also, if e.g. the index of the channel in the vector would be used for e.g. a clusterization implementation afterwards, I think having a new, copied object makes more sense.

Reply:
OK. Let's leave it as is.